lib/MergeStream.js

```diff
@@ -0,0 +1,280 @@
+      this._connectionsStreams[k].destroy();
+    }
+  }
+
+  this.push(null);
+};
+
+MergeStream.prototype._connectionListener = function(connection) {
+  // Load connections until connection is bigger then latest departure time
+
+  if (!connection['@context'] && connection['departureTime'] > this._maxInQueues) {
+    // Pause the stream
+    var streamName = connection['streamName'];
+    var streamIndex = this._connectionsStreamIndex[streamName];
+    // If not ended
+    if (this._connectionsStreams[streamIndex] != null) {
+      this._connectionsStreams[streamIndex].pause();
+    }
+
+    this._read();
+  }
+
+  // Add connection to queues
+  this._addConnection(connection);
+};
+
+// Adds connection to queues
+MergeStream.prototype._addConnection = function (connection) {
+  var added = false;
+
+  // If not a context
+  if (!connection['@context']) {
+    var i = 0;
+    while (!added && i<this._queues.length) {
+      var length = this._queues[i].length;
+
+      // If the queue is empty or connection departureTime is later then last connection in the queue
+      if (length == 0 || this._queues[i][length - 1]['departureTime'] <= connection['departureTime']) {
+        this._queues[i].push(connection);
+        added = true;
+      }
+      i++;
+    }
+  }
+
+  if (!added) {
+    // add new queue
+    var newQueue = [ connection ];
+    this._queues.push(newQueue);
+  }
+
+  // Update maximum departure time in queue
+  if (connection['departureTime'] > this._maxInQueues) {
+    this._maxInQueues = connection['departureTime'];
+  }
+};
+
+
+MergeStream.prototype._getContextIfAvailable = function() {
+  var i = 0;
+  while (i <= this._queues.length) {
+    if (this._queues[i] && this._queues[i].length > 0 && this._queues[i][0]['@context']) {
+      return this._queues[i].shift();
+    }
+    i++;
+  }
+};
+
+MergeStream.prototype._streamsPaused = function() {
+  for (var i=0; i<this._connectionsStreams.length; i++) {
+    if (this._connectionsStreams[i] != null && this._amountStreams == this._amountConnectionsInQueues.length && !this._connectionsStreams[i].isPaused()) {
+      return false;
+    }
+  }
+  return true;
+};
+
+// Every active stream must have connections in the queues
+MergeStream.prototype._allActiveStreamsAvailable = function() {
+  for (var streamName in this._amountConnectionsInQueues) {
+    var amountOfConnectionsInQueue = this._amountConnectionsInQueues[streamName];
+    if (amountOfConnectionsInQueue == 0) {
+      if (Object.keys(this._amountConnectionsInQueues).length == this._amountStreams) {
+        return false;
+      } else {
+        // The stream has ended
+        var removeStreamIndex = this._connectionsStreamIndex[streamName];
+        this._connectionsStreams[removeStreamIndex] = null;
+        delete this._amountConnectionsInQueues[streamName];
+      }
+    }
+  }
+  return true;
+}
+
+/**
+ * Add name of stream to connections
+ */
+function AddStreamNameTransformer (name) {
```

Comment:
It's not because it's a separate file that you have to expose it in the csa.js file :)